examples/depth.rs

```diff
@@ -0,0 +1,193 @@
+                    &wgpu::SwapChainDescriptor {
+                        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+                        format: render_format,
+                        width: size.width.round() as u32,
+                        height: size.height.round() as u32,
+                        present_mode: wgpu::PresentMode::Vsync,
+                    },
+                );
+
+                depth_view = create_depth_view(&device, size);
+            }
+            _ => {}
+        });
+
+        // Get a command encoder for the current frame
+        let mut encoder =
+            device.create_command_encoder(&wgpu::CommandEncoderDescriptor {
+                todo: 0,
+            });
+
+        // Get the next frame
+        let frame = swap_chain.get_next_texture();
+
+        // Clear frame
+        {
+            let _ = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                color_attachments: &[
+                    wgpu::RenderPassColorAttachmentDescriptor {
+                        attachment: &frame.view,
+                        resolve_target: None,
+                        load_op: wgpu::LoadOp::Clear,
+                        store_op: wgpu::StoreOp::Store,
+                        clear_color: wgpu::Color {
+                            r: 0.4,
+                            g: 0.4,
+                            b: 0.4,
+                            a: 1.0,
+                        },
+                    },
+                ],
+                depth_stencil_attachment: None,
+            });
+        }
+
+        // Queue text on top, it will be drawn first.
+        // Depth buffer will make it appear on top.
+        glyph_brush.queue(Section {
+            screen_position: (30.0, 30.0),
+            text: "On top",
+            scale: Scale::uniform(95.0),
+            color: [0.8, 0.8, 0.8, 1.0],
+            z: 0.9,
+            ..Section::default()
+        });
+
+        // Queue background text next.
+        // Without a depth buffer, this text would be rendered on top of the
+        // previous queued text.
+        glyph_brush.queue(Section {
+            bounds: (size.width as f32, size.height as f32),
+            text: &include_str!("lipsum.txt").replace("\n\n", "").repeat(10),
+            scale: Scale::uniform(30.0),
+            color: [0.05, 0.05, 0.1, 1.0],
+            z: 0.2,
+            ..Section::default()
+        });
+
+        // Draw all the text!
+        glyph_brush.draw_queued(
+            &mut device,
+            &mut encoder,
+            &frame.view,
+            wgpu::RenderPassDepthStencilAttachmentDescriptor {
+                attachment: &depth_view,
+                depth_load_op: wgpu::LoadOp::Clear,
+                depth_store_op: wgpu::StoreOp::Store,
+                stencil_load_op: wgpu::LoadOp::Clear,
+                stencil_store_op: wgpu::StoreOp::Store,
+                clear_depth: -1.0,
+                clear_stencil: 0,
+            },
```

Comment:
Same question here. Could we simply ask for a `TextureView` (i.e. the `depth_view`) here?

Reply:
For me, I would also find it useful to pass in the whole `RenderPassDepthStencilAttachmentDescriptor`. In my case, I'm using a different value for `clear_depth`.